File: frontend/app/features/project/graphql/query/project.ts
export const PROJECT_ID_QUERY = `
  query Project(
    $id: ID!
  ) {
    project(
      id: $id
    ) {
      id
      slug
      title
      description
      demoURL
      role
      duration
      createdAt
      updatedAt
      detail {
        id
        problem
        solution
        responsibilities {
          id
          responsibility
          sortOrder
        }
        results {
          id
          result
          sortOrder
        }
      }
    }
  }
`

File: backend/internal/modules/project/model/project.go
package model

import (
	commonModel "backend/internal/common/model"
	authModel "backend/internal/modules/auth/model"
)

type Project struct {
    commonModel.BaseModel

    UserID uint            `gorm:"not null;index"`
    User   *authModel.User `gorm:"foreignKey:UserID;references:ID"`
    Detail *ProjectDetail  `gorm:"foreignKey:ProjectID;constraint:OnDelete:CASCADE;"`

    Slug        string `gorm:"not null;uniqueIndex"`
    Title       string `gorm:"not null"`
    Description string
    Role        string
    Duration    string
    DemoURL     string
}